kubernetes/resource_kubernetes_endpointslice.go

```diff
@@ -0,0 +1,196 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package kubernetes
+
+import (
+	"context"
+	err "errors"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	api "k8s.io/api/discovery/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	pkgApi "k8s.io/apimachinery/pkg/types"
+)
+
+func resourceKubernetesEndpointSlice() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceKubernetesEndpointSliceCreate,
+		ReadContext:   resourceKubernetesEndpointSliceRead,
+		UpdateContext: resourceKubernetesEndpointSliceUpdate,
+		DeleteContext: resourceKubernetesEndpointSliceDelete,
+
+		Schema: map[string]*schema.Schema{
+			"metadata": namespacedMetadataSchema("endpoint_slice", true),
+			"address_type": {
+				Type:        schema.TypeString,
+				Description: "address_type specifies the type of address carried by this EndpointSlice. All addresses in this slice must be the same type. This field is immutable after creation.",
+				Required:    true,
+				ForceNew:    true,
+				ValidateFunc: func(v interface{}, k string) ([]string, []error) {
+					addressType := v.(string)
+
+					if addressType != "IPv4" && addressType != "IPv6" && addressType != "FQDN" {
+						return nil, []error{err.New("address_type: must be either IPv4, IPv6, or FQDN")}
+					}
+					return nil, nil
+				},
```

Comment:
You don't need to do this, see [StringInSlice](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/helper/validation#StringInSlice). 😄